### DISCUSSION TITLE
Migrating vaultwarden data from docker version

### DISCUSSION THREAD

**Discussion by funky123** [tteck/Proxmox#995] [Help Needed]
I am trying to migrate my exisiting vaultwarden setup to aa new lxc from tteck.

i am using the latest Docker image vaultwarden/server:latest
I shut down the Docker container, copied all data via WinSCP to my Windows machine, then I spun up a new Vaultwarden using a script v5 from tteck (thanks for the easy way!) I logged in as admin and configured at the new Vaultwarden settings. After that, I stopped the service via systemctl, replaced all data from the data-folder (except for the config) and got a 502 error after starting the service again. Is anyone experiencing the same issues?

**Comment by tteck** [comment]
Why not use export/import?

**Reply by tteck** [reply]
![vault](https://user-images.githubusercontent.com/21342670/215295063-e29de6fb-d52a-4cbc-8bc6-2bd7c240b038.png)
![Screenshot 2023-01-28 5 50 24 PM](https://user-images.githubusercontent.com/21342670/215295073-fe3be5b1-eeb0-4860-831b-055b70df299f.png)

**Reply by LilTrublMakr** [reply]
I have multiple users on my instance. While it would work for me if I was alone, I would have to have all of my users do the same and transfer over at the same time.

I tried to just transfer my data folder from the old instance to new (using https://github.com/dani-garcia/vaultwarden/wiki/Backing-up-your-vault) but when restarting the LXC, I get a 502. I'll continue to work on this a bit more, but I think this is along the right way to do a complete migration.

**Comment by funky123** [comment]
Will this feature include all Users Groups an Keys, so that logged in devices will stay logged in?

**Comment by LilTrublMakr** [comment]
Figured it out. Here is what I did:

- Spun up the Vaultwarden LXC on Proxmox
- Logged in as root
- Stopped the Vaultwarden process: `systemctl stop vaultwarden`
- Copied over all config files from the docker container to "/opt/vaultwarden/config" (I used Filezilla but you do you boo)
- Chmod all config files to make sure they are owned by Vaultwarden `chmod -R /opt/vaultwarden/config`
- Started the Vaultwarden process `systemctl start vaultwarden`
- Cleared cache and cookies in my browser (I didn't try to log in before clearing, you might not need to do this.)
  - I was able to log in to the new instance on the browser extension with no problem.
- Smile at a job well done.

**Reply by vember31** [reply]
These instructions were super helpful! A couple of things I found when migrating:

- When copying files, I needed to delete / remove the RSA files or else I wasn't able to log in on the new instance.  Those RSA files regenerate.
- Vaultwarden utilizes, at least by default, the /opt/vaultwarden/data/ directory instead of the /opt/vaultwarden/config directory
- Changing the owner uses the 'chown' command (e.g. chown -R vaultwarden:vaultwarden /opt/vaultwarden/data). This command will ensure all files are accessible by the vaultwarden process.

Otherwise, this was a fully successful transition.  Thank you!
